openshift-kube-apiserver/admission/customresourcevalidation/dns/validate_dns.go

```diff
@@ -142,3 +143,31 @@ func validateTolerations(versionedTolerations []corev1.Toleration, fldPath *fiel
 	allErrors = append(allErrors, apivalidation.ValidateTolerations(unversionedTolerations, fldPath)...)
 	return allErrors
 }
+
+// validateUpstreamResolvers validates configuration of the Upstream objects when TLS is configured.
+func validateUpstreamResolvers(upstreamResolvers operatorv1.UpstreamResolvers, fieldPath *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	tc := operatorv1.DNSTransportConfig{}
+
+	// TransportConfig is empty so mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig == tc {
+		return errs
+	}
+
+	switch upstreamResolvers.TransportConfig.Transport {
```

Comment:
do you want to add a `default:` with panic or "unknown transport" error?

Reply:
ah, ok, not familiar with kubebuiler , sorry

Reply:
The kubebuilder validation will prevent users from putting an incorrect value in, but, the validation can be avoided if a user sets `validation=false` on their CLI, so it's best to always handle unknown values in these switches with an error stating that the value is unknown.

Another reason to add a default case is because new values could be added to the enum at a later date. If this is merged into the CRD schema before this operator knows about it, you'll want to identify that. Having an error being returned should bubble up and become very obvious that this operator doesn't support the new config

Reply:
Thanks @JoelSpeed ! I'll update this.